src/main/java/org/opensearch/index/codec/customcodecs/CustomCodecPlugin.java

```diff
@@ -49,12 +50,19 @@ public Optional<CodecServiceFactory> getCustomCodecServiceFactory(final IndexSet
             || codecName.equals(CustomCodecService.QAT_DEFLATE_CODEC)) {
             return Optional.of(new CustomCodecServiceFactory());
         } else {
-            if (!QatZipperFactory.isQatAvailable()
-                && (codecName.equals(Lucene99QatCodec.Mode.QAT_LZ4.getCodec())
-                    || codecName.equals(Lucene99QatCodec.Mode.QAT_DEFLATE.getCodec()))) {
-                throw new IllegalArgumentException("QAT codecs are not supported. Please create indices with a different codec.");
+            if (codecName.equals(Lucene99QatCodec.Mode.QAT_LZ4.getCodec())
```

Comment:
should we refactor this in a way to avoid multiple such if conditions in the future? Maybe a simple switch case for every QAT codec name?